packages/synapse-react-client/src/components/EntityAclEditor/useNotifyNewACLUsers.test.ts

```diff
@@ -0,0 +1,244 @@
+import { server } from '../../mocks/msw/server'
+import { act, renderHook, waitFor } from '@testing-library/react'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import useNotifyNewACLUsers, {
+  shouldNotifyUserInNewResourceAccess,
+} from './useNotifyNewACLUsers'
+import SynapseClient from '../../synapse-client'
+import { ResourceAccess } from '@sage-bionetworks/synapse-types'
+import {
+  mockUserData1,
+  mockUserData2,
+  mockUserData3,
+} from '../../mocks/user/mock_user_profile'
+import { getAccessTypeFromPermissionLevel } from '../../utils/PermissionLevelToAccessType'
+import { MOCK_ACCESS_TOKEN } from '../../mocks/MockSynapseContext'
+import { MOCK_TEAM_ID, mockTeamUserGroups } from '../../mocks/team/mockTeam'
+import {
+  ANONYMOUS_PRINCIPAL_ID,
+  AUTHENTICATED_PRINCIPAL_ID,
+  PUBLIC_PRINCIPAL_ID,
+} from '../../utils/SynapseConstants'
+import {
+  mockAnonymousUserGroupData,
+  mockAuthenticatedGroupData,
+  mockPublicGroupData,
+} from '../../mocks/usergroup/mockUserGroup'
+
+const subject = 'the message subject'
+const body = 'the message body'
+
+const CURRENT_USER_RESOURCE_ACCESS: ResourceAccess = {
+  principalId: mockUserData1.id,
+  accessType: getAccessTypeFromPermissionLevel('CAN_ADMINISTER'),
+}
+
+const sendMessageSpy = jest.spyOn(SynapseClient, 'sendMessage')
+
+describe('useNotifyNewACLUsers', () => {
+  beforeEach(() => jest.clearAllMocks())
+  beforeAll(() => server.listen())
+  afterEach(() => server.restoreHandlers())
+  afterAll(() => server.close())
+
+  it('Sends a notification to new users', async () => {
```

Comment:
Two basic integration tests to verify 1) the hook sends a notification when it should and 2) it does not when it shouldn't.

Unit tests capture complexity below

Reply:
Excellent set of tests